libs/core/langchain_core/runnables/branch.py

```diff
@@ -264,3 +266,117 @@ async def ainvoke(
             raise
         run_manager.on_chain_end(dumpd(output))
         return output
+
+    def stream(
+        self,
+        input: Input,
+        config: Optional[RunnableConfig] = None,
+        **kwargs: Optional[Any],
+    ) -> Iterator[Output]:
+        """First evaluates the condition,
+        then delegate to true or false branch."""
+        config = ensure_config(config)
+        callback_manager = get_callback_manager_for_config(config)
+        run_manager = callback_manager.on_chain_start(
+            dumpd(self),
+            input,
+            name=config.get("run_name"),
+        )
+        outputs = []
+
+        try:
+            for idx, branch in enumerate(self.branches):
+                condition, runnable = branch
+
+                expression_value = condition.invoke(
+                    input,
+                    config=patch_config(
+                        config,
+                        callbacks=run_manager.get_child(tag=f"condition:{idx + 1}"),
+                    ),
+                )
+
+                if expression_value:
+                    for output in runnable.stream(
+                        input,
+                        config=patch_config(
+                            config,
+                            callbacks=run_manager.get_child(tag=f"branch:{idx + 1}"),
+                        ),
+                        **kwargs,
+                    ):
+                        yield output
+                        outputs.append(output)
```

Comment:
we try to ensure the output is the same when called with or without streaming. to achieve that we combine streamed chunks with `+` instead of accumulating in a list

Reply:
See here for an example https://github.com/langchain-ai/langchain/blob/master/libs/core/langchain_core/runnables/base.py#L2318